braindecode/models/sleep_stager_eldele_2021.py

```diff
@@ -0,0 +1,408 @@
+        self.conv1 = nn.Conv1d(inplanes, planes, stride)
+        self.bn1 = nn.BatchNorm1d(planes)
+        self.relu = nn.ReLU(inplace=True)
+        self.conv2 = nn.Conv1d(planes, planes, 1)
+        self.bn2 = nn.BatchNorm1d(planes)
+        self.se = _SELayer(planes, reduction)
+        self.downsample = downsample
+        self.stride = stride
+        self.features = nn.Sequential(self.conv1, self.bn1, self.relu, self.conv2, self.bn2,
+                                      self.se)
+
+    def forward(self, x):
+        residual = x
+        out = self.features(x)
+
+        if self.downsample is not None:
+            residual = self.downsample(x)
+
+        out += residual
+        out = self.relu(out)
+
+        return out
+
+
+class _MRCNN(nn.Module):
+    def __init__(self, after_reduced_cnn_size, shhs):
+        super(_MRCNN, self).__init__()
+        drate = 0.5
+        self.GELU = nn.GELU()
+        self.features1 = nn.Sequential(
+            nn.Conv1d(1, 64, kernel_size=50, stride=6, bias=False, padding=24),
+            nn.BatchNorm1d(64),
+            self.GELU,
+            nn.MaxPool1d(kernel_size=8, stride=2, padding=4),
+            nn.Dropout(drate),
+
+            nn.Conv1d(64, 128, kernel_size=8, stride=1, bias=False, padding=4),
+            nn.BatchNorm1d(128),
+            self.GELU,
+
+            nn.Conv1d(128, 128, kernel_size=8, stride=1, bias=False, padding=4),
+            nn.BatchNorm1d(128),
+            self.GELU,
+
+            nn.MaxPool1d(kernel_size=4, stride=4, padding=2)
+        )
+        if not shhs:
+            self.features2 = nn.Sequential(
+                nn.Conv1d(1, 64, kernel_size=400, stride=50, bias=False, padding=200),
+                nn.BatchNorm1d(64),
+                self.GELU,
+                nn.MaxPool1d(kernel_size=4, stride=2, padding=2),
+                nn.Dropout(drate),
+
+                nn.Conv1d(64, 128, kernel_size=7, stride=1, bias=False, padding=3),
+                nn.BatchNorm1d(128),
+                self.GELU,
+
+                nn.Conv1d(128, 128, kernel_size=7, stride=1, bias=False, padding=3),
+                nn.BatchNorm1d(128),
+                self.GELU,
+
+                nn.MaxPool1d(kernel_size=2, stride=2, padding=1)
+            )
+        else:
+            self.features2 = nn.Sequential(
+                nn.Conv1d(1, 64, kernel_size=400, stride=50, bias=False, padding=200),
+                nn.BatchNorm1d(64),
+                self.GELU,
+                nn.MaxPool1d(kernel_size=4, stride=2, padding=2),
+                nn.Dropout(drate),
+
+                nn.Conv1d(64, 128, kernel_size=6, stride=1, bias=False, padding=3),
+                nn.BatchNorm1d(128),
+                self.GELU,
+
+                nn.Conv1d(128, 128, kernel_size=6, stride=1, bias=False, padding=3),
+                nn.BatchNorm1d(128),
+                self.GELU,
+
+                nn.MaxPool1d(kernel_size=2, stride=2, padding=1)
+            )
```

Comment:
in any case here it does not become so clear what is the difference, most of the code is the same right? if it is only kernel sizes 6 instead of 7 you could just set these as variables before. Right now really hard to tell what is the difference